In [1]:
import pandas as pd

In [6]:
df = pd.read_csv('EjemploDatos.csv')
print('Nº of rows: ', df.shape[0])
df.head()

Nº of rows:  9


prediction date  10u_(44.0, -9.5)  10u_(44.0, -9.375)  \
0  2016-01-01 00:00:00          1.577097            1.675974   
1  2016-01-01 01:00:00          1.320722            1.059857   
2  2016-01-01 02:00:00          1.562336            1.023518   
3  2016-01-01 03:00:00          0.198002            0.025639   
4  2016-01-01 04:00:00          1.604583            1.043792   

   10u_(44.0, -9.25)  10u_(44.0, -9.125)  10u_(44.0, -9.0)  \
0           1.836863            1.821115          1.681467   
1           1.200238            1.276654          1.713910   
2           0.415607            0.335896          0.563801   
3           0.332890           -0.118526         -0.533199   
4           1.073821            0.065032         -0.961823   

   10u_(44.0, -8.875)  10u_(44.0, -8.75)  10u_(44.0, -8.625)  \
0            1.424021           1.190012            1.106394   
1            1.246137           0.725141            0.203412   
2            0.696369           0.487873            0.106770   
3           -1.093990          -1.123653           -1.040646   
4           -1.394685          -1.394074           -1.230012   

   10u_(44.0, -8.5)  ...  vel100_(42.875, -6.0)  vel100_(43.0, -6.0)  \
0          1.108713  ...               4.542353             5.321404   
1          0.215253  ...               4.708214             5.435709   
2         -0.157024  ...               4.553245             5.325067   
3         -0.928341  ...               4.668390             5.177444   
4         -1.500520  ...               4.791137             5.317643   

   vel100_(43.125, -6.0)  vel100_(43.25, -6.0)  vel100_(43.375, -6.0)  \
0               5.971150              5.862873               5.164530   
1               6.092992              5.361253               3.485183   
2               6.018884              4.991258               2.281320   
3               5.716941              4.493564               1.185751   
4               5.648662              4.902998               3.442938   

   vel100_(43.5, -6.0)  vel100_(43.625, -6.0)  vel100_(43.75, -6.0)  \
0             5.391425               6.107658              6.398007   
1             3.988883               4.845162              5.260955   
2             2.807106               3.654994              4.170403   
3             1.837592               3.329041              4.260316   
4             3.724846               4.111763              4.328406   

   vel100_(43.875, -6.0)  vel100_(44.0, -6.0)  
0               6.472929             6.793221  
1               5.464975             5.908251  
2               4.632105             5.078648  
3               4.934864             5.643699  
4               4.885375             5.730442  

[5 rows x 3481 columns]

In [5]:
df.describe()

10u_(44.0, -9.5)  10u_(44.0, -9.375)  10u_(44.0, -9.25)  \
count          9.000000            9.000000           9.000000   
mean           0.343466           -0.116875          -0.270832   
std            1.404674            1.524782           1.749342   
min           -2.131102           -2.594481          -3.071835   
25%            0.001539           -0.707567          -0.680596   
50%            0.591620            0.025639           0.332890   
75%            1.562336            1.043792           1.073821   
max            1.604583            1.675974           1.836863   

       10u_(44.0, -9.125)  10u_(44.0, -9.0)  10u_(44.0, -8.875)  \
count            9.000000          9.000000            9.000000   
mean            -0.699516         -1.004923           -1.251193   
std              1.936031          2.180241            2.122506   
min             -3.889706         -4.586483           -4.775936   
25%             -1.557061         -2.424249           -2.851739   
50%             -0.118526         -0.961823           -1.305223   
75%              0.335896          0.563801            0.696369   
max              1.821115          1.713910            1.424021   

       10u_(44.0, -8.75)  10u_(44.0, -8.625)  10u_(44.0, -8.5)  \
count           9.000000            9.000000          9.000000   
mean           -1.352294           -1.392170         -1.354464   
std             1.988303            1.811029          1.664325   
min            -4.795467           -4.627987         -4.192196   
25%            -2.967039           -2.780516         -2.367186   
50%            -1.229540           -1.230012         -1.308153   
75%             0.487873            0.106770         -0.157024   
max             1.190012            1.106394          1.108713   

       10u_(44.0, -8.375)  ...  vel100_(42.875, -6.0)  vel100_(43.0, -6.0)  \
count            9.000000  ...               9.000000             9.000000   
mean            -1.143676  ...               4.931986             5.840153   
std              1.587899  ...               0.445986             0.745539   
min             -3.778988  ...               4.542353             5.177444   
25%             -1.895506  ...               4.668390             5.321404   
50%             -1.094408  ...               4.791137             5.435709   
75%             -0.133709  ...               5.059935             6.261467   
max              1.304880  ...               5.980429             7.270196   

       vel100_(43.125, -6.0)  vel100_(43.25, -6.0)  vel100_(43.375, -6.0)  \
count               9.000000              9.000000               9.000000   
mean                6.644941              6.027194               4.129468   
std                 1.030375              1.223523               1.723564   
min                 5.648662              4.493564               1.185751   
25%                 5.971150              4.991258               3.442938   
50%                 6.092992              5.862873               4.044214   
75%                 7.395229              7.293384               5.164530   
max                 8.254039              7.901614               6.295951   

       vel100_(43.5, -6.0)  vel100_(43.625, -6.0)  vel100_(43.75, -6.0)  \
count             9.000000               9.000000              9.000000   
mean              4.793565               5.861476              6.305510   
std               1.800871               1.985498              1.974626   
min               1.837592               3.329041              4.170403   
25%               3.724846               4.111763              4.328406   
50%               5.391425               6.107658              6.272207   
75%               5.829247               7.635651              8.362846   
max               7.033442               8.644081              9.221421   

       vel100_(43.875, -6.0)  vel100_(44.0, -6.0)  
count               9.000000             9.000000  
mean                6.636802        

In [7]:
# number of columns that contain the string 10u
df.columns[df.columns.str.contains('10u')].shape[0]

435

In [8]:
#number of columns that contain the string vel100
df.columns[df.columns.str.contains('vel100')].shape[0]

435

In [21]:
#unique values of the first 4 characters of the column names
df.columns.str[:4].unique()

Index(['pred', '10u_', '10v_', '2t_(', 'sp_(', '100u', '100v', 'vel1'], dtype='object')

In [22]:
#falta diferenciar vel10 y vel 100 
print(8*435+1)

3046
3481


The EjemploDatos.csv contains 3481 columns that can be divided into: 
 - Prediction Date : 
 
 - 10u             :
  
 - 10v             : 
 
 - 2t              : 
 
 - sp              : 
 
 - 100u            :
 
 - 100v            :
 
 - vel10           :
 
 - vel100          : 
 

In [4]:
target_df = pd.read_csv('EjemploTarget.csv')
target_df.head()

2016-01-01 00:00:00   4126.7
0  2016-01-01 01:00:00  5688.51
1  2016-01-01 02:00:00  7742.01
2  2016-01-01 03:00:00  7660.50
3  2016-01-01 04:00:00  9730.48
4  2016-01-01 05:00:00  9771.57